recipes/b2/portable/conanfile.py

```diff
@@ -53,49 +57,115 @@ def validate(self):
             raise ConanInvalidConfiguration(
```

Comment:
The above check should use `self.info.options`



```sh
   compiler.libcxx=libstdc++11
   compiler.version=11
   os=Linux
   
   Profile build:
   [settings]
   arch=x86_64
   build_type=Release
   compiler=gcc
   compiler.libcxx=libstdc++11
   compiler.version=11
   os=Linux
   
   
   -------- Computing dependency graph --------
   Graph root
       virtual
   Requirements
       b2/4.7.2#1af70a8be555cb9754e8ae9d066e06b7 - Cache
   
   -------- Computing necessary packages --------
   ERROR: b2/4.7.2: Error in validate() method, line 56
   	if (self.options.toolset == 'cxx' or self.options.toolset == 'cross-cxx') and not self.options.use_cxx_env:
   	ConanException: 'self.options' access in 'validate()' method is forbidden
```

Reply:
@prince-chrismc sounds like a new bug?

Reply:
Maybe 🙃